File: native/ios/BrieflySpeechModule.swift
import Foundation
import Speech

@objc(BrieflySpeechModule)
final class BrieflySpeechModule: NSObject {
  private var startedAt: Date?

  @objc
  func isAvailable(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    resolve(SFSpeechRecognizer.authorizationStatus() != .denied)
  }

  @objc
  func startTranscription(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    startedAt = Date()
    resolve(nil)
  }

  @objc
  func stopTranscription(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    let duration = Int(Date().timeIntervalSince(startedAt ?? Date()))
    resolve([
      "text": "iOS transcription placeholder from SFSpeechRecognizer bridge",
      "durationSeconds": duration
    ])
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    true
  }
}

File: native/windows/BrieflySpeechModule.cs
using Microsoft.ReactNative.Managed;
using System;
using System.Collections.Generic;

namespace Briefly.Windows
{
    [ReactModule("BrieflySpeechModule")]
    public sealed class BrieflySpeechModule
    {
        private DateTimeOffset _startedAt;

        [ReactMethod("isAvailable")]
        public bool IsAvailable() => true;

        [ReactMethod("startTranscription")]
        public void StartTranscription()
        {
            _startedAt = DateTimeOffset.UtcNow;
        }

        [ReactMethod("stopTranscription")]
        public Dictionary<string, object> StopTranscription()
        {
            var duration = (int)(DateTimeOffset.UtcNow - _startedAt).TotalSeconds;
            return new Dictionary<string, object>
            {
                ["text"] = "Windows transcription placeholder from speech bridge",
                ["durationSeconds"] = duration
            };
        }
    }
}
